src/generic/datavgen.cpp

```diff
@@ -861,10 +862,38 @@ class wxDataViewMainWindow: public wxWindow
     bool HasChildren( unsigned int row ) const;
 
 #if wxUSE_DRAG_AND_DROP
+    enum DropHint
+    {
+        DropHint_None = 0,
+        DropHint_Inside,
+        DropHint_Below,
+        DropHint_Above
+    };
+    struct DropItemInfo
+    {
+        unsigned int        m_row;
+        DropHint            m_hint;
+
+        wxDataViewItem      m_item;
+        int                 m_proposedDropIndex;
+        int                 m_indentLevel;
+
+        DropItemInfo()
+        :   m_row(-1)
```

Comment:
```suggestion
        :   m_row(static_cast<unsigned int>(-1))
```
to avoid MSVS warning about signed/unsigned conversion.

Reply:
No problem!
To prevent such situations on macOS with **clang** compiler (and maybe any Linux system with **gcc/clang** complier) +  cmake build system, CMakeLists must be modified for **clang** and **gcc** compilers:

```
if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
    "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")

    ADD_DEFINITIONS (-Wall)

endif ()
```
etc.

Reply:
This is already done in #1825, but I don't use CMake personally.